runtime/oti/jclprots.h

```diff
@@ -968,22 +968,42 @@ jobject JNICALL Java_java_lang_invoke_MethodType_makeTenured(JNIEnv *env, jclass
 #if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
 void JNICALL Java_java_lang_invoke_MethodHandleNatives_init(JNIEnv *env, jclass clazz, jobject self, jobject ref);
 void JNICALL Java_java_lang_invoke_MethodHandleNatives_expand(JNIEnv *env, jclass clazz, jobject self);
-#if JAVA_SPEC_VERSION >= 16
-jobject JNICALL Java_java_lang_invoke_MethodHandleNatives_resolve(JNIEnv *env, jclass clazz, jobject self, jclass caller, jint lookupMode, jboolean speculativeResolve);
-#else /* JAVA_SPEC_VERSION >= 16 */
-jobject JNICALL Java_java_lang_invoke_MethodHandleNatives_resolve(JNIEnv *env, jclass clazz, jobject self, jclass caller, jboolean speculativeResolve);
-#endif /* JAVA_SPEC_VERSION >= 16 */
-jint JNICALL Java_java_lang_invoke_MethodHandleNatives_getMembers(JNIEnv *env, jclass clazz, jclass defc, jstring matchName, jstring matchSig, jint matchFlags, jclass caller, jint skip, jobjectArray results);
+jobject JNICALL Java_java_lang_invoke_MethodHandleNatives_resolve(
+#if JAVA_SPEC_VERSION == 8
+                                                                  JNIEnv *env, jclass clazz,
+                                                                  jobject self, jclass caller);
+#elif JAVA_SPEC_VERSION == 11 /* JAVA_SPEC_VERSION == 8 */
+                                                                  JNIEnv *env, jclass clazz, jobject self,
+                                                                  jclass caller, jboolean speculativeResolve);
+#elif JAVA_SPEC_VERSION >= 16 /* JAVA_SPEC_VERSION == 11 */
+                                                                  JNIEnv *env, jclass clazz, jobject self,
+                                                                  jclass caller, jint lookupMode, jboolean speculativeResolve);
+#else /* JAVA_SPEC_VERSION == 8 */
+#error Unsupported Java_java_lang_invoke_MethodHandleNatives_resolve
```

Comment:
I don't think we need this line

Reply:
Why? because it already exists in `java_lang_invoke_MethodHandleNatives.cpp`? or we do not need to worry about the gaps created by the `ifdef/else` statements?

Reply:
`#error` will allow us to explicitly fail in the gaps created by the `ifdef/elif` statements. Without the `#error`, the failure will be implicit during runtime, which will be difficult to locate and triage. So, I recommend to keep the `#error`.

Reply:
Historically this have not been done in OpenJ9 to method headers in particular as the gaps between `ifdef/elif` is out of support. I guess this is more of a preference thing as it doesn't have any impact on actual LTS builds. 
If we were to keep this `#error` message, we should only have it in the header file to avoid redundant code?
@tajila any thought on this?

Reply:
Wouldnt the runtime failure be an unsatisfied linkage exception?

Reply:
No, it didn't crash for Java 8. The JVM just ran with the incorrect signature.